library/core/class.controller.php

```diff
@@ -2142,4 +2164,23 @@ public function title($Title = null, $Subtitle = null) {
 
         return $this->data('Title');
     }
+
+    /**
+     * Set the destination URL where the page will be redirected after an ajax request.
+     *
+     * @param string|null $destination Destination URL or path.
+     *      Redirect to current URL if nothing or null is supplied.
+     * @param bool $trustedOnly Non trusted destinations will be redirected to /home/leaving?Target=$destination
+     */
+    public function redirectTo($destination = null, $trustedOnly = true) {
+        if ($destination === null) {
+            $url = url('');
+        } elseif ($trustedOnly) {
+            $url = safeURL($destination);
+        } else {
+            $url = url($destination);
+        }
+
+        $this->redirectTo = $url;
```

Comment:
If an addon sets RedirectUrl it will be honoured: https://github.com/vanilla/vanilla/pull/5703/files#diff-0b51e9233f2e072d92f57f60e061864dR1284